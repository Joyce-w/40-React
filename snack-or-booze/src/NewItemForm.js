import React, { useState } from "react";
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';
import "./NewItemForm.css"
import SnackOrBoozeApi from "./Api";
import Axios from "axios";
import { useHistory} from "react-router-dom";

const NewItemForm = ({ updateState }) => {
    //inital template for snack form
    const initial_item = {
        category: "snack",
        id: "",
        itemName: "",
        description: "",
        recipe: "",
        serve: ""
    }

    const [item, setItem] = useState(initial_item)
    const history = useHistory();

    //update function on current form data
    const handleChange = (e) => {
        const { name, value } = e.target;

        setItem(item => ({
            ...item,
            [name]: value
        }))
    }

    //handle changes when form is submitted
    const handleSubmit=(e) => {
        // e.preventDefault();
        async function postFood(item) {

            let newItem = {
                "id": item.itemName.split(' ').join('-'),
                "name": item.itemName,
                "description": item.description,
                "recipe": item.description,
                "serve": item.serve
            }
            
            if (item.category === 'snack') {
                await SnackOrBoozeApi.postSnacks(newItem)
                updateState('snack', newItem)
            }
            else{
                await SnackOrBoozeApi.postDrinks(newItem)
                updateState('drink', newItem)
            }
        }
        postFood(item)
        history.push('/')
    }

    return (
    <Form className="w-25">
        <FormGroup>
        <Label for="category">Select</Label>
        <Input onChange={ handleChange} type="select" name="category" id="category">
            <option value="snack">Snack</option>
            <option value="drink">Drink</option>
        </Input>
      </FormGroup>
        <FormGroup>
            <Label for="itemName">Name</Label>
                <Input onChange={ handleChange} type="text" name="itemName" id="itemName"  />
        </FormGroup>
        <FormGroup>
            <Label for="description">Description</Label>
            <Input onChange={ handleChange} type="textarea" name="description" id="description"  />
        </FormGroup>
        <FormGroup>
            <Label for="recipe">Recipe</Label>
            <Input onChange={ handleChange} type="textarea" name="recipe" id="recipe" />
        </FormGroup>
        <FormGroup>
            <Label for="serve">Serve</Label>
            <Input onChange={ handleChange} type="textarea" name="serve" id="serve" />
            </FormGroup>
            <Button onClick={handleSubmit }>Submit</Button>
    </Form>
    )
}

export default NewItemForm;