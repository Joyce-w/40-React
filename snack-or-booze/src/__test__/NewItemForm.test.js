import React from 'react'
import { render, fireEvent } from '@testing-library/react'
//use jest to test for jest matchers
import '@testing-library/jest-dom/extend-expect'
import NewItemForm from '../NewItemForm'
import {MemoryRouter} from 'react-router-dom'


//smoke test
it("should render w/o crashing", () => {
    render(<NewItemForm/> )
})

//snapshot test
it("should match snapshot", () =>{
  const {asFragment} = render(<NewItemForm/>)
  expect(asFragment()).toMatchSnapshot();

})

//specialized testing
it("should display form", () => {
  const {getByLabelText, getByText} = render(<NewItemForm/>)
  expect(getByLabelText('Select')).toBeInTheDocument();
  expect(getByLabelText('Description')).toBeInTheDocument();
  expect(getByText('Submit')).toBeInTheDocument();
})

