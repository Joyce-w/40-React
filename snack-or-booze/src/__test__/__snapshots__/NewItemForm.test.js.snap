// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="w-25"
  >
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="category"
      >
        Select
      </label>
      <select
        class="form-control"
        id="category"
        name="category"
      >
        <option
          value="snack"
        >
          Snack
        </option>
        <option
          value="drink"
        >
          Drink
        </option>
      </select>
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="itemName"
      >
        Name
      </label>
      <input
        class="form-control"
        id="itemName"
        name="itemName"
        type="text"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="description"
      >
        Description
      </label>
      <textarea
        class="form-control"
        id="description"
        name="description"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="recipe"
      >
        Recipe
      </label>
      <textarea
        class="form-control"
        id="recipe"
        name="recipe"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="serve"
      >
        Serve
      </label>
      <textarea
        class="form-control"
        id="serve"
        name="serve"
      />
    </div>
    <button
      class="btn btn-secondary"
      type="button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
